sympy/sets/sets.py

```diff
@@ -1298,8 +1298,8 @@ def identity(self):
     def zero(self):
         return S.UniversalSet
 
-    def __new__(cls, *args, **kwargs):
-        evaluate = kwargs.get('evaluate', global_parameters.evaluate)
+    def __new__(cls, *args , evaluate=global_parameters.evaluate):
+        
```

Comment:
Parameter default values are evaluated at function definition time so this will store the value of `global_parameters.evaluate` at the time that the `def` statement is executed (when this module is imported). The purpose of the `global_parameters.evaluate` flag is that it can be changed dynamically but using as a default parameter value like this will not work as expected if someone later changes the value.

Reply:
It should be more like
```python
def __new__(cls, *args, evaluate=None):
    if evaluate is None:
        evaluate = global_parameters.evaluate
```

Reply:
ok will do it